pyg_lib/ops/__init__.py

```diff
@@ -40,7 +40,11 @@ def grouped_matmul(inputs: List[Tensor], others: List[Tensor],
     if int(major_vers) >= 2 or int(minor_vers) >= 14:
         inputs = torch.nested.as_nested_tensor(inputs).contiguous()
         others = torch.nested.as_nested_tensor(others).contiguous()
-        outs = torch.bmm(inputs, others).contiguous()
+        if inputs.dim() == 4 or others.dim() == 4:
+            # bmm only works on lists of 2D tensors
+            outs = torch.matmul(inputs, others).contiguous()
```

Comment:
Do `matmul` and `bmm` use different code-paths? Is it slower to use `matmul`? Can you shed some light here?

Reply:
i believe the matmul option is using a slower code path under the hood, but since i needed to deal w/ 3D tensor lists for HGTConv it was the only option. if this is undesirable, then i would have to do HGTConv w/ segment matmul only, which requires padding under the hood. thoughts on which you would like to see? if you'd like i can try to benchmark the torch >= 1.14 grouped matmul path agaisnt the torch < 1.14 segment matmul path to see how their speed compares. that can inform our decision

Reply:
unless you see a way to shape tensors [here](https://github.com/pyg-team/pytorch_geometric/blob/hgtconv_pyglib/torch_geometric/nn/conv/hgt_conv.py#L298-L327) to be a list of 2D tensors instead of 3D. I tried a few ways but ended up with the wrong shapes or some other issue

Reply:
Never really thought about this use-case. I am happy to merge this now, and potentially revisit the internal code path later.